src/app/patient/create-patient/similar-patient-dialog.ts

```diff
@@ -0,0 +1,147 @@
+            next: response => {
+                let similarityScore = response.data[0].similarityScore;
+                if (similarityScore > 0.8) {
+                    this.resultsFound = true;
+                    let id = response.data[0].biobankId ?? ""
+                    this.loadPatient(id);
+                } else {
+                    this.resultsFound = false;
+                }
+            }
+        }
+        )
+
+    }
+    goToPatientIdCard(match : Patient) {
+        this.dialogRef.close();
+        this.router.navigate(["/idcard",match.ids[0].idType, match.getIdString(match.ids[0].idType)]).then();
+    }
+
+    loadMatches() {
+        console.log('getCheckMatch called');
+        return this.sessionService.createToken("checkMatch", new AddPatientTokenData(
+            ["biobankId"]
+        )).pipe(
+            mergeMap(token => {
+                console.log('Token received:', token);
+                return this.resolveCheckMatch(token.id);
+            }),
+            catchError((error) => {
+                console.error('Error occurred in getCheckMatch:', error);
+                if (error.status >= 400 && error.status < 500) {
+                    return throwError(() => new Error("failed to fetch matches"));
+                } else {
+                    return throwError(() => new Error("failed to fetch matches"));
+                }
+            })
+        );
+    }
+    resolveCheckMatch(tokenId: string | undefined) {
+        let patient: Patient = new Patient();
+        patient.fields = {
+            vorname: "Alf",
+            nachname: "Klemt",
+            geburtsname: "Salz",
+            geburtsjahr: "1989",
+            geburtsmonat: "08",
+            geburtstag: "02"
+        }
+        let body = new URLSearchParams();
+        for (const name in patient.fields) {
+            body.set(name, patient.fields[name]);
+        }
+        return this.httpClient.post(this.patientList.url + "/patients/checkMatch/" + tokenId, body, {
+            headers: new HttpHeaders()
+                .set('Content-Type', 'application/x-www-form-urlencoded')
+                .set('mainzellisteApiVersion', '3.2')
+        }).pipe(
+            mergeMap(response => {
+                type checkMatch = {
+                    biobankId?: string;
+                    similarityScore: number;
+                };
+
+                let result: checkMatch[] = response as checkMatch[];
+                return of({ data: result });
+            }),
+            catchError((error) => {
+                console.error('Error occurred in getCheckMatch:', error);
+                if (error.status >= 400 && error.status < 500) {
+                    return throwError(() => new Error("failed to fetch matches"));
+                } else {
+                    return throwError(() => new Error("failed to fetch matches"));
+                }
+            })
+        );
+
+    }
+
+    loadPatient(idString: string) {
```

Comment:
Diese Methode existiert schon in **patient-list.service.ts**. Sieh ID-Card sie wird dort verwendet